pallets/grants/src/lib.rs

```diff
@@ -315,7 +323,15 @@ pub mod pallet {
 
 				T::Currency::resolve_creating(who, T::Currency::issue(total_grants));
 				T::Currency::set_lock(VESTING_LOCK_ID, who, total_grants, WithdrawReasons::all());
-				<VestingSchedules<T>>::insert(who, schedules);
+
+				match <BoundedVec<VestingScheduleOf<T>, T::MaxSchedule>>::try_from(schedules.clone()) {
```

Comment:
Similar to what Eliot suggested, can we collect the schedules of someones grants in BoundedVec rather than Vec at line 305? If so, then this step of converting vec to bounded vec could be avoided.

Reply:
Can I have bit more details here ?

Reply:
Resolving this, as there has been changes to this file.